eclipse-cdt-cloud/vscode-memory-inspector
# Support storing and applying of memory content

#### What it does
Add commands to store and apply memory content as Intel HEX file
- Encapsulate behavior in new MemoryStorage class
- Trigger 'store' from Memory view, Variables view and command palette
- Trigger 'apply' from Memory view, Explorer view and command palette
- Use nrf-intel-hex library for read/write file licensed under BSD-3

Use quick inputs to guide user through necessary input
- Initialize as much of the input as possible through command args

Communicate with webview through messenger requests and notifications
- Request to trigger store and apply from webview
- Notify webview about any written memory so it can update properly

Minor improvements
- Move some common types and functionality into 'common' area
- Avoid bleeding Debug Adapter types into webview, use messaging types
- Common style: 'getVariables' -> 'getVariablesType'
- Provide utility functions and types for debug requests
- Fix 'Enter' handling for numpad by checking key value of event

https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/assets/19170971/9c3e2772-cfd5-4b7c-8d45-e2707338d0cb


Closes #50

#### How to test
- Run a debug session
- Check how you can store memory (from the inspector, from the command, from the Variables window)
- Try applying your stored memory (from the inspector, from the file context menu, from the file explorer context menu)

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
